[android] style: enhance theme to match the green theme of om

kavikhalique opened the pull request:
Fix: #7590 
**Enhanced the UI and Matched the theme of OM Android app with the theme of Organic map**
- The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green

**Issue Solved**
- Greenified the whole theme accoridng to OM
- Enhanced Night Mode Theme with theme of Organic green theme.

<img src="https://github.com/organicmaps/organicmaps/assets/120750626/732deb93-85e9-4e4f-8943-a54d4fe1d7e5" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/e7ede056-d20b-4eea-ab26-1f3f66a960f4" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/94b3108f-701d-4885-8e0d-7ff55d77fea5" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/3ff52b89-2a5b-4092-81fb-5405dd8e496b" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/2e2ceaa5-6944-480a-80ac-cb80c0803f8a" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/f02981ee-f072-450d-acb1-1bd7c0a60e1b" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/54f3cb89-9ebd-499b-964a-7320a862e024" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/1364081f-ab48-490d-810f-55d61b46173b" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/b5735ddc-cd88-48ca-9c19-40e04f9eaedd" width="300">

https://github.com/organicmaps/organicmaps/assets/120750626/6f1c8b60-4b17-48dd-b07b-f2ac236c03c2

vng commented:
Thanks, looks fresh, IMO!

Misalf-git commented:
But, please, not the map screen buttons and things that weren't blue before.

kavikhalique commented:
> But, please, not the map screen buttons and things that weren't blue before.

Previous grey color was giving feeling of disabled and less interactive according to the actual greenish theme, i have just spread the green color little bit so that it provide a green organic feel.

kavikhalique commented:
> That's way too much green imo. I would keep the colors of the buttons as they are.

Please suggest where do you feel it is too much greenish. I will try to make it better.

Misalf-git commented:
These:

:--: | :--: | :--: | :--:
![312981218-f02981ee-f072-450d-acb1-1bd7c0a60e1b.jpg](https://github.com/organicmaps/organicmaps/assets/72605561/4f23f764-dc3b-483b-872d-a05cac8aedd3) | ![312981241-1364081f-ab48-490d-810f-55d61b46173b.jpg](https://github.com/organicmaps/organicmaps/assets/72605561/b160b352-e86f-4b1b-9b5d-e9a11a5ec6dc) | ![312979595-94b3108f-701d-4885-8e0d-7ff55d77fea5.jpg](https://github.com/organicmaps/organicmaps/assets/72605561/2727e8c3-ecf3-4dd5-aa7f-bc065f1a6052) | ![312979612-3ff52b89-2a5b-4092-81fb-5405dd8e496b.jpg](https://github.com/organicmaps/organicmaps/assets/72605561/f6f98720-cb9a-4f73-906b-009bd02ee3b2)

kavikhalique commented:
> That's way too much green imo. I would keep the color of the buttons as they are.

Ok, will try to make it look less greenish and more better.
Please suggest if you have any idea it will be very helpful.

meenbeese commented:
@kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. Also, the redesigned folder icons are subpar as they are too large now.

biodranik commented:
Blue color meant something actionable/clickable. Now it mixes with the background, unclickable green.

Misalf-git commented:
> Also, the redesigned folder icons are subpar as they are too large now.

Oh, I didn't notice the changed icons.
The "download-arrow" icon does look better / more familiar, though.

kavikhalique commented:
> @kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. Also, the redesigned folder icons are subpar as they are too large now.

Will try to do some better changes to make it look better. Thankyou for your suggestions.

euf commented:
This is a welcome change! 

Do you mind if we tweak colors of some elements a little more? 
If that's okay, in a few days I'm gonna suggest separate colors to be used for…

- Splash screen background;
- Menu header background;
- Accent color for the menu buttons;
- Accent color for on-the map buttons.

Based on the brand book UI color scheme: https://github.com/organicmaps/organicmaps/wiki/Colors

Judging from your screenshots, these adjustments may further improve consistency and contrast.

kavikhalique commented:
> This is a welcome change! 
> 
> 
> 
> Do you mind if we tweak colors of some elements a little more? 
> 
> If that's okay, in a few days I'm gonna suggest separate colors to be used for…
> 
> 
> 
> - Splash screen background;
> 
> - Menu header background;
> 
> - Accent color for the menu buttons;
> 
> - Accent color for on-the map buttons.
> 
> 
> 
> Based on the brand book UI color scheme: https://github.com/organicmaps/organicmaps/wiki/Colors
> 
> 
> 
> Judging from your screenshots, these adjustments may further improve consistency and contrast.

I would love to work on this.

pastk commented:
> @kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. 

Could you please link to your previous PR?

pastk commented:
> * The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green

TBH it never jumped to my eye as some inconsistency :/ Maybe I've just used to it over the years, of course :)
I don't see an issue having a few different colors in the UI (as long they're used consistently, of course).

Maybe the currently used blue color is not the best choice for actionable controls, indeed. However to my taste now everything looks **too green and too monotonous**.

Looks like quite some users from the chat also prefer the current grey maps buttons:
![image](https://github.com/organicmaps/organicmaps/assets/18434508/d582f6d1-c938-4add-a3b6-834f955de77e)

Looking forward for more ideas as e.g. @euf suggested!

kavikhalique commented:
> > * The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green
> 
> 
> 
> TBH it never jumped to my eye as some inconsistency :/ Maybe I've just used to it over the years, of course :)
> 
> I don't see an issue having a few different colors in the UI (as long they're used consistently, of course).
> 
> 
> 
> Maybe the currently used blue color is not the best choice for actionable controls, indeed. However to my taste now everything looks **too green and too monotonous**.
> 
> 
> 
> Looks like quite some users from the chat also prefer the current grey maps buttons:
> 
> ![image](https://github.com/organicmaps/organicmaps/assets/18434508/d582f6d1-c938-4add-a3b6-834f955de77e)
> 
> 
> 
> Looking forward for more ideas as e.g. @euf suggested!
> 
> 
> 
> 

The first thing that strike my mind after opening the app was the colour. Although present theme is not bad but i feel there should be a systematic use of colours not only green but any combination of colours should be used in a proper combination and systematic way (as said by @euf as-well).

While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.

I am looking forward to contribute some positive changes to OM.

biodranik commented:
> While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.

Where is it inconsistent in your opinion? Blue means something clickable (except main map and bottom toolbar buttons), green is usually a background color.

kavikhalique commented:
> > While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.
> 
> Where is it inconsistent in your opinion? Blue means something clickable (except main map and bottom toolbar buttons), green is usually a background color.

**Here blue colour has been used for text theme and its no clickable**
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/88a57c62-53ff-4541-ac7e-071551635bf1" width="300">

**Here report a bug is clickable but its not blue**
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/0c7cefb7-0fd1-4b5b-a33f-6841bf9a01ec" width="300">

**Here blue colour was just used to show enable and disable**
<img src="https://github.com/organicmaps/organicmaps/assets/120750626/fc28b288-5b61-4d1a-b30b-357dea043258" width="300">

IMO instead of using blue as clickable we can use green as well. Whatsapp also have green theme and they use green for both clickable and disabled as well.
Blue can also be used in case of OM but at someplaces blue and green should have used in combination as well to give a feeling that both colors are systematically chosen.

biodranik commented:
Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.

Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

kavikhalique commented:
> Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.
> 
> Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

yes it can also be done. But if we are selecting dual colour theme we should bring blue colour in themes of OM as well IMO.

kavikhalique commented:
> Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.
> 
> 
> 
> Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

IMO here green one looks better i guess

![image](https://github.com/organicmaps/organicmaps/assets/120750626/ca8c60fb-6c34-413f-8ae8-eced32e08742)

previous one

![image](https://github.com/organicmaps/organicmaps/assets/120750626/31a8adaa-7703-48f2-961d-83875357247d)

biodranik commented:
The Blue color is a familiar, attractive one. It is used in many other apps to attract user's attention to click the most important button. Check other maps apps for example.

kavikhalique commented:
> If that's okay, in a few days I'm gonna suggest separate colors to be used for…
>
>
It will be very helpful if you guide me through it because at this point i am really very confused what should i choose.

In my opinion, We can do either of the three things 

1 - let the UI be as it is, just replace green at places where it is un clickable but blue in colour as suggested by @biodranik 

2 - redesign UI properly considering both green and blue as a part of the brand colour and select theme which gives effect of both colour

3 - implement a simple toggle button in preferences to let the users choose between either monotonic theme or green tone theme.

biodranik commented:
Start with 1, and think about 2. Definitely not 3.

euf commented:
I agree, definitely not **3**. 

At the moment I'm working on **2**, trying to figure out the newest "Material 3" design approach. For building color schemes it uses second and third accent colors in addition to a primary one. I was hoping that would simply translate to our green vs. blue approach, but things turned out to be more complicated. Further tinkering is needed, and now I'm not sure it's going to work out soon. 

This leaves us with **1** if we are ready to move incrementally. Do you mind sharing screenshots of this approach?

kavikhalique commented:
> Start with 1, and think about 2. Definitely not 3.

Ok for now will start with 1 and with suggestions will improve it

kavikhalique commented on android/app/src/main/res/color/accent_color_selector_night.xml:
```diff
@@ -1,6 +1,6 @@
+  <item android:color="@color/bg_night_primary"
         android:state_selected="true"/>
 
   <item android:color="@color/text_light_subtitle"/>
```
Sorry if i am understanding wrong but do you mean that we should not add .png files instead we should add other format or you mean we should not add any more image resources to the project.

RedAuburn replied:
> Sorry if i am understanding wrong but do you mean that we should not add .png files instead we should add other format or you mean we should not add any more image resources to the project.

i think he means it'd be better to use SVG 👍️

kavikhalique replied:
Oh thankyou, understood

kavikhalique commented:
> What is the current status here? 

Since there are a lot of confusions. I have gave it a pause since various distinct opinions are coming which are opposite of counter of each others.

Jean-BaptisteC commented:
I suggest for this PR to keep only changes in light theme.
Update dark theme colors and icons can be done later.
@kavikhalique can you rebase your branch?

kavikhalique commented:
> I suggest for this PR to keep only changes in light theme.
> 
> Update dark theme colors and icons can be done later.
> 
> @kavikhalique can you rebase your branch and apply suggestions?

Sure i will do that

kavikhalique commented:
- Replaced png icons with svg files
- Removed Green icons from bottom sheet
- Changed the night mode to skyblue

@Jean-BaptisteC 

Here is the video of UI


https://github.com/organicmaps/organicmaps/assets/120750626/d0bef130-8589-4827-8437-7fabcc43e89a

Jean-BaptisteC commented:
I prefer buttons in grey

euf commented:
I'm not 100% sure about the green accent on the map buttons, but let's try and see how this works IRL. 
Replacing PNGs with SVGs would definitely allow more flexibility in further adjustments of UI colors 👍

Misalf-git commented:
For me, in the light style, color on the buttons make them less "readable". In the dark style they pop out too much.

kavikhalique commented:
According to past few suggestions i think its better to revert back the button color to grey in light mode and white in dark mode.

@rtsisyk @Jean-BaptisteC @euf

biodranik commented:
Changing important action colors in the app should be carefully planned and designed. It would be great to first design it properly and consistently with our long-term plans (e.g. new Place Page, new Navigation UI), and change colors only after that.

P.S. Compare the importance of this issue/change (and time spent on it) with some other issues, crashes, bugs, missing features, etc.

biodranik commented:
My opinion:
1. There are a lot of more important issues to fix than this one. The app works fine without this change.
2. It would also be great to spend reviewers' time more efficiently, e.g. by focusing on the most impactful/critical issues.
3. The chosen color is controversial and not clear.
4. Changing colors is tricky and may break the UX/complicate the code
5. Changing colors now will highly likely require changing them again later after the app redesign. So it would be better to postpone it.

kavikhalique commented:
After various reviews and suggestions, right now its not correct time to change the UI so closing this PR.
